awapi/awapi_awiv_adapter/include/awapi_awiv_adapter/awapi_autoware_state_publisher.h

```diff
@@ -14,63 +14,60 @@
  * limitations under the License.
  */
 
-#include <autoware_api_msgs/AwapiAutowareStatus.h>
+#include <rclcpp/rclcpp.hpp>
+
 #include <awapi_awiv_adapter/awapi_autoware_util.h>
+#include <autoware_api_msgs/msg/awapi_autoware_status.hpp>
 
 namespace autoware_api
 {
-class AutowareIvAutowareStatePublisher
+class AutowareIvAutowareStatePublisher : public rclcpp::Node
```

Comment:
Only `AutowareIvAdapter` class should be inheriting the `rclcpp::Node` class.
Otherwise, it will create another node.
(Same goes for `AutowareIvVehicleStatePublisher`, `AutowareIvStopReasonAggregator`, `AutowareIvLaneChangeStatePublisher`, AutowareIvObstacleAvoidanceStatePublisher`)

If you need access to the loggers or create pub/sub within this class, then we should pass `AutowareIvAdapter`'s loggers and topic_interface  (e.g. by `get_node_topics_interface ()`) or just pass `AutowareIvAdapter`'s node pointer to each class.

Reply:
No problem, I can do that. I assumed it would be possible to create nested nodes, since the ROS1 code also had several `NodeHandles`, but I tested with a toy example and it doesn't work. I'm asking an expert in Apex.AI if there's a way to make it work, just for completeness.

Reply:
Just making sure: I was informed that having nested nodes is possible if you add them to the executor, but you'd prefer there to be only one node, right?

Reply:
Since original ROS1 code is in one node, I think we want to keep it as one during porting.